### DISCUSSION TITLE
This project have both rtmp and libdash  library But i am fail to understand rtmp to dash packaging

### DISCUSSION THREAD

**Discussion by chandra9651** [ireader/media-server#241] [Q&A] [answered]
I am searching project for rtmp to dash packaging then I got this great project and try to understand rtmp to dash packaging flow but can not clearly understand this flow . I want to understand this flow because I am  interesting this project and help to improve it and make better it 
Please any one help to better understanding

**Comment by ireader** [comment, marked as answer]
To use rtmp with dash,  see more libdash/test/dash-dynamic-test.cpp.

At function dash_live_worker, a while loop load local file as dash input.  You can change it to rtmp server callback (onvideo/onaudio), e.g. librtmp/test/rtmp-server-publish-test.cpp

**Reply by chandra9651** [reply]
Thankyou chen

**Reply by chandra9651** [reply]
Chen my main problem is
  
                  rtmp stream            how we do this  ?                hls or dash
Publisher send  --------------------------->  Server ( rtmp to hls or dsah  )-----------------------------------> viewer

Convertion of rtmp to hls on Server this is main problem for me  
Please Help Chen

**Reply by chandra9651** [reply]
Chen  can you provide documentation of this for better understanding of code flow
Because I facing problem to reading program

**Reply by ireader** [reply]
Streaming server,  usually has a source stream and many sink stream.  Source stream and sink stream are connected by dispatcher.

1. rtmp source stream read from network -> cache ring buffer;
2. dispatch notify sink stream to read;
3. hls/dash sink stream read from buffer -> muxer ts/mp4 segment -> send;

This is a simplest rtmp server: librtmp/test/rtmp-server-forward-aio-test.cpp.

**Reply by chandra9651** [reply]
Does dispatcher is file or function in this project and i can not find buffer where rtmp write stream , please provide locations
also provide  cache ring buffer - location in file
